components/contribution-flow/index.js

```diff
@@ -171,12 +172,21 @@ class ContributionFlow extends React.Component {
         variables: {
           order: {
             quantity: stepDetails.quantity,
-            amount: { valueInCents: stepDetails.amount },
+            amount:
+              this.props.paymentMethod === 'crypto'
+                ? { value: stepDetails.amount }
+                : { valueInCents: stepDetails.amount },
             frequency: getGQLV2FrequencyFromInterval(stepDetails.interval),
             guestInfo,
             fromAccount,
             toAccount: pick(this.props.collective, ['id']),
-            customData: stepDetails.customData,
+            customData:
+              this.props.paymentMethod === 'crypto'
+                ? {
+                    pledgeAmount: stepDetails.amount,
+                    pledgeCurrency: stepDetails.currency.value,
```

Comment:
Why not store this in `customData` in the first place?

Reply:
I thought about this. Note that if we do that, we will have to store in both places in `stepDetails.amount`, `stepDetails.currency.value` and on `customData: {pledgeAmount, pledgeCurrency}` because the contribution flow relies on the `stepDetails.amount` and `stepDetails.currency` and those values i think should be there; 

https://github.com/opencollective/opencollective-frontend/blob/e58f925bf1b69c8bb4a09ba9399c6c4284e06fd7/components/contribution-flow/index.js#L775

Note that the above function relies on the amount value for example. I didn't want to include these values both in the `stepDetails.amount` and `stepDetails.customData` (similarly for currency) which is why I thought of doing it this way. Let me know your thoughts.